Firestore/Source/Local/FSTMutationQueue.h

```diff
@@ -115,10 +116,21 @@ NS_ASSUME_NONNULL_BEGIN
  * don't contain the document key at all if it's convenient.
  */
 // TODO(mcg): This should really return an NSEnumerator
-// also for b/32992024, all backing stores should really index by document key
 - (NSArray<FSTMutationBatch *> *)allMutationBatchesAffectingDocumentKey:
```

Comment:
Question: perhaps this function can now be removed or reimplemented in terms of the new one? (I can do that if you think it's a good idea)

Reply:
I believe the other can be removed since it's an anti-pattern. I purely went with the addition of this new method here to give you the chance to figure out how to integrate it into the reworked higher-level algorithm.

Reply:
Ok, I'll do that in a follow-up.